### PR TITLE
Use custom transport for remote targets

### PR DESCRIPTION
**Pull request by poornas**
## Motivation and Context
Remove response timeout for remote targets used in replication

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation updated
- [ ] Unit tests added/updated

### CODE REVIEW

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :x: [more...](#1184e78c-3abe-4279-bfda-c04ce9f9616a) |
<a name="1184e78c-3abe-4279-bfda-c04ce9f9616a"></a>
### 12080-db61e95/mint-compress-encrypt-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev7.minio.io:30238
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 5d880665897a:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 43 seconds
(4/15) Running aws-sdk-ruby tests ... done in 3 seconds
(5/15) Running awscli tests ... FAILED in 32 seconds
{
  "name": "awscli",
  "duration": 2675,
  "function": "aws --endpoint-url http://minio-dev7.minio.io:30238 s3api copy-object --bucket awscli-mint-test-bucket-4289 --key datafile-1-kB-copy --copy-source awscli-mint-test-bucket-4289/datafile-1-kB\n",
  "status": "FAIL",
  "error": "Hash mismatch expected 084e1383b70fb0c51acc680fef370023, got ac57de7156d7fc25ac1a65f81fa3989b"
}
(5/15) Running healthcheck tests ... done in 0 seconds
(6/15) Running mc tests ... done in 46 seconds
(7/15) Running minio-dotnet tests ... done in 41 seconds
(8/15) Running minio-go tests ... FAILED in 1 minutes and 54 seconds
{
  "args": {},
  "duration": 827,
  "error": "At least one of the pre-conditions you specified did not hold",
  "function": "CopyObjectPart(destination, source)",
  "message": "CopyObjectPart call failed",
  "name": "minio-go: testUnencryptedToSSES3CopyObjectPart",
  "status": "FAIL"
}
(8/15) Running minio-java tests ... FAILED in 1 minutes and 27 seconds
{
  "name": "minio-java",
  "function": "copyObject()",
  "args": "[match etag]",
  "duration": 128,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = PreconditionFailed, message = At least one of the pre-conditions you specified did not hold, bucketName = minio-java-test-1dk2o64, objectName = minio-java-test-3gbvqhb-copy, resource = /minio-java-test-1dk2o64/minio-java-test-3gbvqhb-copy, requestId = 16767603580E70B4, hostId = 32b12c05-c4bf-40ac-8181-15e1848c058a)\nrequest={method=PUT, url=http://minio-dev7.minio.io:30238/minio-java-test-1dk2o64/minio-java-test-3gbvqhb-copy, headers=x-amz-copy-source-if-match: 71cff0a060f852067e443ad1e24ae26c-1\nx-amz-copy-source: /minio-java-test-2kdedak/minio-java-test-3gbvqhb\nHost: minio-dev7.minio.io:30238\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.0.3\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20210416T220752Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20210416/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-copy-source;x-amz-copy-source-if-match;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=412, headers=Accept-Ranges: bytes\nContent-Length: 418\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nETag: \"71cff0a060f852067e443ad1e24ae26c\"\nLast-Modified: Fri, 16 Apr 2021 22:07:52 GMT\nServer: MinIO\nVary: Origin\nX-Amz-Request-Id: 16767603580E70B4\nX-Xss-Protection: 1; mode=block\nDate: Fri, 16 Apr 2021 22:07:52 GMT\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:775), io.minio.MinioClient.execute(MinioClient.java:563), io.minio.MinioClient.executePut(MinioClient.java:904), io.minio.MinioClient.copyObject(MinioClient.java:1232), FunctionalTest.testCopyObjectMatchETag(FunctionalTest.java:1850), FunctionalTest.copyObject(FunctionalTest.java:2016), FunctionalTest.runObjectTests(FunctionalTest.java:3757), FunctionalTest.runTests(FunctionalTest.java:3783), FunctionalTest.main(FunctionalTest.java:3927)]"
}
(8/15) Running minio-js tests ... done in 49 seconds
(9/15) Running minio-py tests ... done in 2 minutes and 54 seconds
(10/15) Running s3cmd tests ... FAILED in 5 seconds
{
  "name": "s3cmd",
  "duration": "2670",
  "function": "test_put_object_multipart",
  "status": "FAIL",
  "error": "WARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Too many failures. Giving up on '/mint/data/datafile-65-MB'\nERROR: \nUpload of '/mint/data/datafile-65-MB' part 1 failed. Use\n  /usr/local/bin/s3cmd abortmp s3://s3cmd-test-bucket-25110/s3cmd-test-object-12434 aa6b0963-7cc3-4d07-8321-ebaa42d95c71\nto abort the upload, or\n  /usr/local/bin/s3cmd --upload-id aa6b0963-7cc3-4d07-8321-ebaa42d95c71 put ...\nto continue the upload.\nERROR: Upload of '/mint/data/datafile-65-MB' failed too many times (Last reason: )"
}
(10/15) Running s3select tests ... done in 6 seconds
(11/15) Running security tests ... done in 0 seconds

Executed 11 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Review by harshavardhana — APPROVED** [review]

File: cmd/utils.go [hunk 1]
```diff
@@ -637,6 +638,34 @@ func newGatewayHTTPTransport(timeout time.Duration) *http.Transport {
...
+// NewRemoteTargetHTTPTransport returns a new http configuration
+// used while communicating with the remote replication targets.
+func NewRemoteTargetHTTPTransport() *http.Transport {
+	// For more details about various values used here refer
+	// https://golang.org/pkg/net/http/#Transport documentation
+	tr := &http.Transport{
+		Proxy: http.ProxyFromEnvironment,
+		DialContext: (&net.Dialer{
+			Timeout:   30 * time.Second,
+			KeepAlive: 30 * time.Second,
```
Comment (harshavardhana) [inline cmd/utils.go]: Perhaps keep this to 15 @poornas

File: cmd/utils.go [hunk 2]
```diff
@@ -637,6 +638,34 @@ func newGatewayHTTPTransport(timeout time.Duration) *http.Transport {
...
+		}).DialContext,
+		MaxIdleConnsPerHost:   1024,
+		WriteBufferSize:       16 << 10, // 16KiB moving up from 4KiB default
+		ReadBufferSize:        16 << 10, // 16KiB moving up from 4KiB default
+		IdleConnTimeout:       30 * time.Second,
```
Comment (harshavardhana) [inline cmd/utils.go]: Keep this 15?

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: cmd/utils.go [hunk 3]
```diff
@@ -637,6 +638,34 @@ func newGatewayHTTPTransport(timeout time.Duration) *http.Transport {
...
 
+// NewRemoteTargetHTTPTransport returns a new http configuration
+// used while communicating with the remote replication targets.
+func NewRemoteTargetHTTPTransport() *http.Transport {
+	// For more details about various values used here refer
+	// https://golang.org/pkg/net/http/#Transport documentation
+	tr := &http.Transport{
+		Proxy: http.ProxyFromEnvironment,
+		DialContext: (&net.Dialer{
+			Timeout:   30 * time.Second,
```
Comment (harshavardhana) [inline cmd/utils.go]:
```suggestion
			Timeout:   15 * time.Second,
```

File: cmd/utils.go [hunk 4]
```diff
@@ -637,6 +638,34 @@ func newGatewayHTTPTransport(timeout time.Duration) *http.Transport {
...
+			KeepAlive: 15 * time.Second,
```
Comment (harshavardhana) [inline cmd/utils.go]:
```suggestion
			KeepAlive: 30 * time.Second,
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#dd90cd38-c145-4b40-9047-f5dd6a05c36a) |
| mint-compress-encrypt-dist-erasure.sh | :x: [more...](#e6b50233-6812-439e-a7ce-54a45755b5f8) |
<a name="dd90cd38-c145-4b40-9047-f5dd6a05c36a"></a>
### 12080-e8507ed/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c2.minio.io:31900
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp f3341a011a73:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 42 seconds
(4/15) Running aws-sdk-ruby tests ... done in 4 seconds
(5/15) Running awscli tests ... done in 2 minutes and 19 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... FAILED in 38 seconds
{
  "name": "mc",
  "duration": "10834",
  "function": "test_mirror_list_objects_storage_class",
  "status": "FAIL",
  "error": "mirror and list differs"
}
(7/15) Running minio-dotnet tests ... done in 51 seconds
(8/15) Running minio-go tests ... FAILED in 1 minutes and 32 seconds
{
  "args": {
    "bucketName": "minio-go-test-j2zkmpkfqwjj3owk",
    "objectName": "wersqyosaf1ent4o3zc253m3il9ryr"
  },
  "duration": 9365,
  "error": "read tcp 172.17.0.8:34094->72.28.97.58:31900: read: connection reset by peer",
  "function": "GetObject(bucketName, objectName)",
  "message": "ReadAt failed",
  "name": "minio-go: testSSES3EncryptedGetObjectReadAtFunctional",
  "status": "FAIL"
}
(8/15) Running minio-java tests ... done in 1 minutes and 2 seconds
(9/15) Running minio-js tests ... done in 45 seconds
(10/15) Running minio-py tests ... done in 2 minutes and 33 seconds
(11/15) Running s3cmd tests ... done in 17 seconds
(12/15) Running s3select tests ... done in 6 seconds
(13/15) Running security tests ... done in 1 seconds

Executed 13 out of 15 tests successfully.
```
<a name="e6b50233-6812-439e-a7ce-54a45755b5f8"></a>
### 12080-e8507ed/mint-compress-encrypt-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev8.minio.io:32384
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 3d9c6e642090:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 43 seconds
(4/15) Running aws-sdk-ruby tests ... done in 4 seconds
(5/15) Running awscli tests ... FAILED in 32 seconds
{
  "name": "awscli",
  "duration": 2777,
  "function": "aws --endpoint-url http://minio-dev8.minio.io:32384 s3api copy-object --bucket awscli-mint-test-bucket-31589 --key datafile-1-kB-copy --copy-source awscli-mint-test-bucket-31589/datafile-1-kB\n",
  "status": "FAIL",
  "error": "Hash mismatch expected 084e1383b70fb0c51acc680fef370023, got ac57de7156d7fc25ac1a65f81fa3989b"
}
(5/15) Running healthcheck tests ... done in 0 seconds
(6/15) Running mc tests ... done in 49 seconds
(7/15) Running minio-dotnet tests ... done in 42 seconds
(8/15) Running minio-go tests ... FAILED in 1 minutes and 50 seconds
{
  "args": {},
  "duration": 517,
  "error": "At least one of the pre-conditions you specified did not hold",
  "function": "CopyObjectPart(destination, source)",
  "message": "CopyObjectPart call failed",
  "name": "minio-go: testUnencryptedToSSES3CopyObjectPart",
  "status": "FAIL"
}
(8/15) Running minio-java tests ... FAILED in 1 minutes and 32 seconds
{
  "name": "minio-java",
  "function": "copyObject()",
  "args": "[match etag]",
  "duration": 270,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = PreconditionFailed, message = At least one of the pre-conditions you specified did not hold, bucketName = minio-java-test-15j1krt, objectName = minio-java-test-2v8qmij-copy, resource = /minio-java-test-15j1krt/minio-java-test-2v8qmij-copy, requestId = 16767BCFAB4EF932, hostId = 0160d7e5-a6e9-45a0-a13c-15b3de250c49)\nrequest={method=PUT, url=http://minio-dev8.minio.io:32384/minio-java-test-15j1krt/minio-java-test-2v8qmij-copy, headers=x-amz-copy-source-if-match: 71cff0a060f852067e443ad1e24ae26c-1\nx-amz-copy-source: /minio-java-test-1e2een8/minio-java-test-2v8qmij\nHost: minio-dev8.minio.io:32384\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.0.3\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20210416T235407Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20210416/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-copy-source;x-amz-copy-source-if-match;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=412, headers=Accept-Ranges: bytes\nContent-Length: 418\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nETag: \"71cff0a060f852067e443ad1e24ae26c\"\nLast-Modified: Fri, 16 Apr 2021 23:54:07 GMT\nServer: MinIO\nVary: Origin\nX-Amz-Request-Id: 16767BCFAB4EF932\nX-Xss-Protection: 1; mode=block\nDate: Fri, 16 Apr 2021 23:54:07 GMT\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:775), io.minio.MinioClient.execute(MinioClient.java:563), io.minio.MinioClient.executePut(MinioClient.java:904), io.minio.MinioClient.copyObject(MinioClient.java:1232), FunctionalTest.testCopyObjectMatchETag(FunctionalTest.java:1850), FunctionalTest.copyObject(FunctionalTest.java:2016), FunctionalTest.runObjectTests(FunctionalTest.java:3757), FunctionalTest.runTests(FunctionalTest.java:3783), FunctionalTest.main(FunctionalTest.java:3927)]"
}
(8/15) Running minio-js tests ... done in 53 seconds
(9/15) Running minio-py tests ... done in 2 minutes and 59 seconds
(10/15) Running s3cmd tests ... FAILED in 5 seconds
{
  "name": "s3cmd",
  "duration": "2795",
  "function": "test_put_object_multipart",
  "status": "FAIL",
  "error": "WARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Too many failures. Giving up on '/mint/data/datafile-65-MB'\nERROR: \nUpload of '/mint/data/datafile-65-MB' part 1 failed. Use\n  /usr/local/bin/s3cmd abortmp s3://s3cmd-test-bucket-20035/s3cmd-test-object-31102 1281bd00-2fc8-43dd-b10d-cc950fff11f3\nto abort the upload, or\n  /usr/local/bin/s3cmd --upload-id 1281bd00-2fc8-43dd-b10d-cc950fff11f3 put ...\nto continue the upload.\nERROR: Upload of '/mint/data/datafile-65-MB' failed too many times (Last reason: )"
}
(10/15) Running s3select tests ... done in 6 seconds
(11/15) Running security tests ... done in 0 seconds

Executed 11 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Review by kannappanr — APPROVED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :x: [more...](#7819c0d5-ef90-43f0-a49b-8b3c9c7f8e39) |
<a name="7819c0d5-ef90-43f0-a49b-8b3c9c7f8e39"></a>
### 12080-e0227a3/mint-compress-encrypt-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c3.minio.io:30320
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp adab4efd790a:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 43 seconds
(4/15) Running aws-sdk-ruby tests ... done in 3 seconds
(5/15) Running awscli tests ... FAILED in 32 seconds
{
  "name": "awscli",
  "duration": 2394,
  "function": "aws --endpoint-url http://minio-c3.minio.io:30320 s3api copy-object --bucket awscli-mint-test-bucket-10475 --key datafile-1-kB-copy --copy-source awscli-mint-test-bucket-10475/datafile-1-kB\n",
  "status": "FAIL",
  "error": "Hash mismatch expected 084e1383b70fb0c51acc680fef370023, got ac57de7156d7fc25ac1a65f81fa3989b"
}
(5/15) Running healthcheck tests ... done in 0 seconds
(6/15) Running mc tests ... done in 46 seconds
(7/15) Running minio-dotnet tests ... done in 40 seconds
(8/15) Running minio-go tests ... FAILED in 1 minutes and 50 seconds
{
  "args": {},
  "duration": 823,
  "error": "At least one of the pre-conditions you specified did not hold",
  "function": "CopyObjectPart(destination, source)",
  "message": "CopyObjectPart call failed",
  "name": "minio-go: testUnencryptedToSSES3CopyObjectPart",
  "status": "FAIL"
}
(8/15) Running minio-java tests ... FAILED in 1 minutes and 32 seconds
{
  "name": "minio-java",
  "function": "copyObject()",
  "args": "[match etag]",
  "duration": 491,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = PreconditionFailed, message = At least one of the pre-conditions you specified did not hold, bucketName = minio-java-test-3r0cqid, objectName = minio-java-test-30iae3c-copy, resource = /minio-java-test-3r0cqid/minio-java-test-30iae3c-copy, requestId = 16767E270D43F5F1, hostId = a2ebd479-832d-4f24-a54a-b618bed27d99)\nrequest={method=PUT, url=http://minio-c3.minio.io:30320/minio-java-test-3r0cqid/minio-java-test-30iae3c-copy, headers=x-amz-copy-source-if-match: 71cff0a060f852067e443ad1e24ae26c-1\nx-amz-copy-source: /minio-java-test-20t3a9k/minio-java-test-30iae3c\nHost: minio-c3.minio.io:30320\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.0.3\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20210417T003702Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20210417/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-copy-source;x-amz-copy-source-if-match;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=412, headers=Accept-Ranges: bytes\nContent-Length: 418\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nETag: \"71cff0a060f852067e443ad1e24ae26c\"\nLast-Modified: Sat, 17 Apr 2021 00:37:02 GMT\nServer: MinIO\nVary: Origin\nX-Amz-Request-Id: 16767E270D43F5F1\nX-Xss-Protection: 1; mode=block\nDate: Sat, 17 Apr 2021 00:37:02 GMT\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:775), io.minio.MinioClient.execute(MinioClient.java:563), io.minio.MinioClient.executePut(MinioClient.java:904), io.minio.MinioClient.copyObject(MinioClient.java:1232), FunctionalTest.testCopyObjectMatchETag(FunctionalTest.java:1850), FunctionalTest.copyObject(FunctionalTest.java:2016), FunctionalTest.runObjectTests(FunctionalTest.java:3757), FunctionalTest.runTests(FunctionalTest.java:3783), FunctionalTest.main(FunctionalTest.java:3927)]"
}
(8/15) Running minio-js tests ... done in 53 seconds
(9/15) Running minio-py tests ... done in 2 minutes and 54 seconds
(10/15) Running s3cmd tests ... FAILED in 4 seconds
{
  "name": "s3cmd",
  "duration": "2750",
  "function": "test_put_object_multipart",
  "status": "FAIL",
  "error": "WARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Too many failures. Giving up on '/mint/data/datafile-65-MB'\nERROR: \nUpload of '/mint/data/datafile-65-MB' part 1 failed. Use\n  /usr/local/bin/s3cmd abortmp s3://s3cmd-test-bucket-17090/s3cmd-test-object-24552 8bfbbf7b-55f1-46a2-97c6-dbe10f3f7086\nto abort the upload, or\n  /usr/local/bin/s3cmd --upload-id 8bfbbf7b-55f1-46a2-97c6-dbe10f3f7086 put ...\nto continue the upload.\nERROR: Upload of '/mint/data/datafile-65-MB' failed too many times (Last reason: )"
}
(10/15) Running s3select tests ... done in 6 seconds
(11/15) Running security tests ... done in 0 seconds

Executed 11 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**